File: week6/credit.py
# Librarys
from cs50 import get_int
import sys


def main():
    # Get credit card number
    number = get_int("Credit card number: ")
    
    # checking number long enough
    if number < 100000000000:
        print("INVALID")
        sys.exit(1)
    
    # If number long enough
    else:
        # Usefull variables
        n = number % 10
        Nn = number - n
        N = Nn / 10
        Nu = number
        num = number
        y = 0
        x = 0
    
    # Double and add the digits of every second digit starting from the second last
    while N > 0:
        a = (N % 10) * 2
        if a > 9:
            b = a % 10
            c = (a - b) / 10
            y = y + b + c
        else:
            y = a + y
        N = ((N - (N % 100)) / 100)
    
    # Add every digit from the last    
    while Nu > 0:
        d = Nu % 10
        x = d + x
        Nu = ((Nu - (Nu % 100)) / 100)
        
    # Checking sum
    s = x + y
    Csum = s % 10
    
    if Csum != 0:
        print(INVALID)
        sys.exit(1)
    else:
        # Determining which card type
        while num > 9:
            num = ((num - (num % 10)) / 10)
            if 39 < num and num < 50:
                print("VISA")
                sys.exit(0)
            elif 50 < num and num < 56:
                print("MASTERCARD")
                sys.exit(0)
            elif num == 34 or num == 37:
                print("AMEX")
                sys.exit(0)
            elif num < 10:
                print("INVALID")
                sys.exit(0)
            

main()

File: week1/credit.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // TODO: Prompt for credit card number

    long number = get_long("Credit card number: ");

    if (number < 100000000000)
    {
        printf("INVALID\n");
        return 0;
    }

    else if (number > 100000000000)
        // calculate checksum
    {
        //Usefull variables for adding every second digit from the second last and oubling ect
        int n = number % 10;
        long Nn = number - n;
        long N = Nn / 10;

        int a = 0;
        int b = 0;
        int c = 0;
        int y = 0;
        
        //Usefull variables for adding every digit from the last
        long Nu = number;
        int d = 0;
        int x = 0;
        //Double and add the digits of every second digit starting from the second last
        for (N = Nn / 10; N > 0; N = ((N - (N % 100)) / 100))
        {
            a = (N % 10) * 2;
            if (a > 9)
            {
                b = a % 10;
                c = (a - b) / 10;
                y = y + b + c;
            }
            else if (a <= 9)
            {
                y = a + y;
            }
        }

        //Add every digit from the last
        for (Nu = number; Nu > 0; Nu = ((Nu - (Nu % 100)) / 100))
        {
            d = Nu % 10;

            x = d + x;
        }
        
        // Checking sum
        int sum = x + y;
        int Csum = sum  % 10;
        if (Csum != 0)
        {
            printf("INVALID\n");
            return 0;
        }
        else if (Csum == 0)
        {
            // determinint which card type
            int L2 = 0;
            for (Nu = number; Nu > 10; Nu = ((Nu - (Nu % 10)) / 10))
            {
                L2 = Nu;
            }
            if (39 < L2 && L2 < 50)
            {
                printf("VISA\n");
                return 0;
            }
            else if (50 < L2 && L2 < 56)
            {
                printf("MASTERCARD\n");
                return 0;
            }
            else if (L2 == 34 || L2 == 37)
            {
                printf("AMEX\n");
                return 0;
            }
            else
            {
                printf("INVALID\n");
                return 0;
            }
        }
    }
}
